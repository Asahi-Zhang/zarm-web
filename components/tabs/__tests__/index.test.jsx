import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tabs from '../index';

const { Tab } = Tabs;

describe('Tabs', () => {
  it('renders line tabs correctly', () => {
    const wrapper = render(
      <div>
        <Tabs type="line">
          <Tab title="Tab1">
            <div style={{ padding: 10 }}>
              这是选项卡1的文字
            </div>
          </Tab>
          <Tab disabled title="Tab2">
            <div style={{ padding: 10 }}>
              这是选项卡2的文字
            </div>
          </Tab>
          <Tab title="Tab3">
            <div style={{ padding: 10 }}>
              这是选项卡3的文字
            </div>
          </Tab>
          <Tab title="Tab4">
            <div style={{ padding: 10 }}>
              这是选项卡4的文字
            </div>
          </Tab>
        </Tabs>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders card tabs correctly', () => {
    const wrapper = render(
      <div>
        <Tabs type="card">
          <Tab title="Tab1">
            <div style={{ padding: 10 }}>
              这是选项卡1的文字
            </div>
          </Tab>
          <Tab disabled title="Tab2">
            <div style={{ padding: 10 }}>
              这是选项卡2的文字
            </div>
          </Tab>
          <Tab title="Tab3">
            <div style={{ padding: 10 }}>
              这是选项卡3的文字
            </div>
          </Tab>
          <Tab title="Tab4">
            <div style={{ padding: 10 }}>
              这是选项卡4的文字
            </div>
          </Tab>
        </Tabs>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders noborder-card tabs correctly', () => {
    const wrapper = render(
      <div>
        <Tabs type="noborder-card">
          <Tab title="Tab1">
            <div style={{ padding: 10 }}>
              这是选项卡1的文字
            </div>
          </Tab>
          <Tab disabled title="Tab2">
            <div style={{ padding: 10 }}>
              这是选项卡2的文字
            </div>
          </Tab>
          <Tab title="Tab3">
            <div style={{ padding: 10 }}>
              这是选项卡3的文字
            </div>
          </Tab>
          <Tab title="Tab4">
            <div style={{ padding: 10 }}>
              这是选项卡4的文字
            </div>
          </Tab>
        </Tabs>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders tabs size correctly', () => {
    const wrapper = render(
      <div>
        <Tabs type="line" size="sm">
          <Tab title="Tab1">
            <div style={{ padding: 10 }}>
              这是选项卡1的文字
            </div>
          </Tab>
          <Tab disabled title="Tab2">
            <div style={{ padding: 10 }}>
              这是选项卡2的文字
            </div>
          </Tab>
          <Tab title="Tab3">
            <div style={{ padding: 10 }}>
              这是选项卡3的文字
            </div>
          </Tab>
          <Tab title="Tab4">
            <div style={{ padding: 10 }}>
              这是选项卡4的文字
            </div>
          </Tab>
        </Tabs>
        <Tabs type="line" size="md">
          <Tab title="Tab1">
            <div style={{ padding: 10 }}>
              这是选项卡1的文字
            </div>
          </Tab>
          <Tab disabled title="Tab2">
            <div style={{ padding: 10 }}>
              这是选项卡2的文字
            </div>
          </Tab>
          <Tab title="Tab3">
            <div style={{ padding: 10 }}>
              这是选项卡3的文字
            </div>
          </Tab>
          <Tab title="Tab4">
            <div style={{ padding: 10 }}>
              这是选项卡4的文字
            </div>
          </Tab>
        </Tabs>
        <Tabs type="line" size="lg">
          <Tab title="Tab1">
            <div style={{ padding: 10 }}>
              这是选项卡1的文字
            </div>
          </Tab>
          <Tab disabled title="Tab2">
            <div style={{ padding: 10 }}>
              这是选项卡2的文字
            </div>
          </Tab>
          <Tab title="Tab3">
            <div style={{ padding: 10 }}>
              这是选项卡3的文字
            </div>
          </Tab>
          <Tab title="Tab4">
            <div style={{ padding: 10 }}>
              这是选项卡4的文字
            </div>
          </Tab>
        </Tabs>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders line horizontal scroll tabs correctly', () => {
    const wrapper = render(
      <div>
        <Tabs type="line" direction="horizontal" defaultValue={0}>
          {
            [...Array(40).keys()].map((item, index) => (
              <Tab key={index.toString()} title={`Tab${index + 1}`} style={{ padding: 10 }}>
                这是选项卡{index + 1}的文字
              </Tab>
            ))
          }
        </Tabs>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders line vertical scroll tabs correctly', () => {
    const wrapper = render(
      <div>
        <Tabs type="line" closable direction="vertical" defaultValue={0} style={{ height: '192px' }}>
          {
            [...Array(40).keys()].map((item, index) => (
              <Tab key={index.toString()} title={`Tab${index + 1}`}>
                这是选项卡{index + 1}的文字
              </Tab>
            ))
          }
        </Tabs>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
