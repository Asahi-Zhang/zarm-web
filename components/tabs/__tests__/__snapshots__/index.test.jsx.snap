// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders card tabs correctly 1`] = `
<div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--md zw-tabs--card"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item zw-tabs__header__item--disabled"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        <div
          style="padding:10px"
        >
          这是选项卡1的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡2的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡3的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡4的文字
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs renders line horizontal scroll tabs correctly 1`] = `
<div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--md zw-tabs--line"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab5
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab6
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab7
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab8
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab9
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab10
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab11
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab12
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab13
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab14
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab15
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab16
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab17
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab18
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab19
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab20
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab21
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab22
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab23
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab24
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab25
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab26
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab27
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab28
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab29
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab30
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab31
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab32
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab33
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab34
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab35
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab36
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab37
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab38
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab39
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab40
          </div>
          <div
            class="zw-tabs__header__line"
            style="width:0;height:0;transform:translate3d(0px,0,0)"
          />
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
        style="padding:10px"
      >
        这是选项卡1的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡2的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡3的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡4的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡5的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡6的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡7的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡8的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡9的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡10的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡11的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡12的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡13的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡14的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡15的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡16的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡17的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡18的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡19的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡20的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡21的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡22的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡23的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡24的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡25的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡26的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡27的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡28的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡29的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡30的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡31的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡32的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡33的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡34的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡35的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡36的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡37的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡38的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡39的文字
      </div>
      <div
        class="zw-tabs__body__item"
        style="padding:10px"
      >
        这是选项卡40的文字
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs renders line tabs correctly 1`] = `
<div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--md zw-tabs--line"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item zw-tabs__header__item--disabled"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
          <div
            class="zw-tabs__header__line"
            style="width:0;height:0;transform:translate3d(0px,0,0)"
          />
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        <div
          style="padding:10px"
        >
          这是选项卡1的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡2的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡3的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡4的文字
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs renders line vertical scroll tabs correctly 1`] = `
<div>
  <div
    class="zw-tabs zw-tabs--vertical zw-tabs--md"
    style="height:192px"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab5
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab6
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab7
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab8
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab9
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab10
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab11
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab12
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab13
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab14
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab15
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab16
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab17
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab18
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab19
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab20
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab21
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab22
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab23
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab24
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab25
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab26
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab27
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab28
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab29
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab30
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab31
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab32
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab33
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab34
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab35
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab36
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab37
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab38
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab39
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab40
          </div>
          <div
            class="zw-tabs__header__line"
            style="width:0;height:0;transform:translate3d(0,0px,0)"
          />
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        这是选项卡1的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡2的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡3的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡4的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡5的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡6的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡7的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡8的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡9的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡10的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡11的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡12的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡13的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡14的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡15的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡16的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡17的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡18的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡19的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡20的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡21的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡22的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡23的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡24的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡25的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡26的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡27的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡28的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡29的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡30的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡31的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡32的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡33的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡34的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡35的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡36的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡37的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡38的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡39的文字
      </div>
      <div
        class="zw-tabs__body__item"
      >
        这是选项卡40的文字
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs renders noborder-card tabs correctly 1`] = `
<div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--md zw-tabs--noborder-card"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item zw-tabs__header__item--disabled"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        <div
          style="padding:10px"
        >
          这是选项卡1的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡2的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡3的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡4的文字
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs renders tabs size correctly 1`] = `
<div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--sm zw-tabs--line"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item zw-tabs__header__item--disabled"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
          <div
            class="zw-tabs__header__line"
            style="width:0;height:0;transform:translate3d(0px,0,0)"
          />
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        <div
          style="padding:10px"
        >
          这是选项卡1的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡2的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡3的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡4的文字
        </div>
      </div>
    </div>
  </div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--md zw-tabs--line"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item zw-tabs__header__item--disabled"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
          <div
            class="zw-tabs__header__line"
            style="width:0;height:0;transform:translate3d(0px,0,0)"
          />
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        <div
          style="padding:10px"
        >
          这是选项卡1的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡2的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡3的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡4的文字
        </div>
      </div>
    </div>
  </div>
  <div
    class="zw-tabs zw-tabs--horizontal zw-tabs--lg zw-tabs--line"
  >
    <div
      class="zw-tabs__header"
    >
      <div
        class="zw-tabs__header__scroll"
      >
        <div
          class="zw-tabs__header__nav"
        >
          <div
            class="zw-tabs__header__item zw-tabs__header__item--active"
          >
            Tab1
          </div>
          <div
            class="zw-tabs__header__item zw-tabs__header__item--disabled"
          >
            Tab2
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab3
          </div>
          <div
            class="zw-tabs__header__item"
          >
            Tab4
          </div>
          <div
            class="zw-tabs__header__line"
            style="width:0;height:0;transform:translate3d(0px,0,0)"
          />
        </div>
      </div>
    </div>
    <div
      class="zw-tabs__body zw-tabs__body--animated"
      style="margin-left:-0%"
    >
      <div
        class="zw-tabs__body__item zw-tabs__body__item--active"
      >
        <div
          style="padding:10px"
        >
          这是选项卡1的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡2的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡3的文字
        </div>
      </div>
      <div
        class="zw-tabs__body__item"
      >
        <div
          style="padding:10px"
        >
          这是选项卡4的文字
        </div>
      </div>
    </div>
  </div>
</div>
`;
